modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -226,11 +229,28 @@ public void clusterChanged(ClusterChangedEvent event) {
     static boolean hasAtLeastOneGeoipProcessor(ClusterState clusterState) {
         List<PipelineConfiguration> pipelineDefinitions = IngestService.getPipelines(clusterState);
         return pipelineDefinitions.stream().anyMatch(pipelineDefinition -> {
+            if (isLazyDownloadPipeline(pipelineDefinition) && (isPipelineUsed(clusterState, pipelineDefinition.getId()) == false)) {
```

Comment:
ℹ️ For all pipeline that have `_meta.geoip_database_lazy_download` set to `false`, the download is triggered only when an index with the pipeline set as `default_pipeline` or `final_pipeline` exists.